src/Altinn.App.Core/Infrastructure/Clients/Notifications/Email/EmailNotificationClient.cs

```diff
@@ -1,58 +1,44 @@
+using Altinn.App.Core.Internal.Notifications.Email;
+using Altinn.App.Core.Models.Notifications.Email;
 using Altinn.Common.AccessTokenClient.Services;
 using Microsoft.ApplicationInsights;
+using Microsoft.Extensions.Logging;
 using Microsoft.Extensions.Options;
-using Prometheus;
-using System.Diagnostics;
 using System.Net.Http.Headers;
 using System.Text.Json;
 
-namespace Altinn.App.Core.Infrastructure.Clients.Email;
-/// <summary>
-/// Implementation of the <see cref="IEmailNotificationClient"/> interface using a HttpClient to send
-/// requests to the Email Notification service.
-/// </summary>
-public sealed class EmailNotificationClient : IEmailNotificationClient
+namespace Altinn.App.Core.Infrastructure.Clients.Notifications.Email;
+
+internal sealed class EmailNotificationClient : IEmailNotificationClient
 {
+    private readonly ILogger<EmailNotificationClient> _logger;
     private readonly IHttpClientFactory _httpClientFactory;
     private readonly IAppMetadata _appMetadata;
     private readonly PlatformSettings _platformSettings;
     private readonly IAccessTokenGenerator _accessTokenGenerator;
-    private readonly TelemetryClient _telemetryClient;
-    private static readonly Counter _orderCount = Metrics
-        .CreateCounter("altinn_app_notification_order_request_count", "Number of notification order requests.", labelNames: ["type", "result"]);
+    private readonly TelemetryClient? _telemetryClient;
 
-    /// <summary>
-    /// Initializes a new instance of the <see cref="EmailNotificationClient"/> class.
-    /// </summary>
-    /// <param name="httpClientFactory"></param>
-    /// <param name="platformSettings">Api endpoints for platform services.</param>
-    /// <param name="appMetadata">The service providing appmetadata.</param>
-    /// <param name="accessTokenGenerator">An access token generator to create an access token.</param>
-    /// <param name="telemetryClient">Client used to track dependencies.</param>
     public EmailNotificationClient(
+        ILogger<EmailNotificationClient> logger,
         IHttpClientFactory httpClientFactory,
         IOptions<PlatformSettings> platformSettings,
         IAppMetadata appMetadata,
         IAccessTokenGenerator accessTokenGenerator,
-        TelemetryClient telemetryClient)
+        TelemetryClient? telemetryClient = null)
     {
+        _logger = logger;
         _platformSettings = platformSettings.Value;
         _httpClientFactory = httpClientFactory;
         _appMetadata = appMetadata;
         _accessTokenGenerator = accessTokenGenerator;
         _telemetryClient = telemetryClient;
     }
 
-    /// <inheritdoc/>
-    /// <exception cref="EmailNotificationException"/>
     public async Task<EmailOrderResponse> Order(EmailNotification emailNotification, CancellationToken ct)
     {
-        var startTime = DateTime.UtcNow;
-        var timer = Stopwatch.StartNew();
+        using var dependency = new Telemetry.Dependency(_telemetryClient);
```

Comment:
I didn't really intend for it to be used by any other classes than the notification clients, so that's why I didn't parameterize it by name or anything like that.

I'm suprised you categorize `using` as magic, it's a pretty standard part of .NET dev vocabulary isn't it? I only made it to not duplicate the control flow across those two clients, since the App Insights API is a bit bad as you note